### ISSUE TITLE
Interpreter is too slow (at least in emulator)

### ISSUE THREAD

**Issue by Timendus** [Timendus/thumby-silicon8#1]
Ideas to try:

* Get rid of `planeBuffer` to `display` conversion overhead by having the `draw` function just operate on a single plane (or maybe two, but not technically eight)
* Compile C code to RP2040 (or Python to Python bytecode..?) and include that as an `.mpy` file ([link](https://docs.micropython.org/en/latest/develop/natmod.html), [link](https://docs.micropython.org/en/latest/reference/mpyfiles.html#mpy-files))
* Compile the original Silicon8 Go code to RP2040 [using TinyGo](https://tinygo.org/docs/reference/microcontrollers/machine/feather-rp2040/) and build the interpreter that way, or include that as an `.mpy` file. But I haven't yet been able to find any information on the latter.

These are all ideas to play with when I receive the actual Thumby hardware. `.mpy` modules require a file system, which the emulator currently does not have. Also, I don't currently have an accurate idea of just how fast the hardware really is.

**Comment by Timendus** [comment]
Working on the first point in #2, but the new Thumby API came in between. So now the PR is doing multiple things. Ah well...
It can quite effectively work with two `FrameBuffers`. I don't think rendering will be the issue anymore after I'm done with that PR. And if it is, I can easily reduce the frame rate from 60FPS to 30FPS.

Also, I didn't mention another obvious improvement: to use the `@micropython.native` and `@micropython.viper` annotations. My first test with `@micropython.viper` is showing a significant improvement over regular code. I plan on adding these annotations in many places in the interpreter internals, which should improve the number of cycles per frame we can get quite a bit.

In short, I just wanted to leave an update here that I think I'll be able to get this running at an acceptable speed. At least as fast as the original Cosmac VIP, possibly faster.

**Comment by Timendus** [comment]
The thing is now faster than the Cosmac VIP in my estimate (on hardware). So I'm closing this issue 😉
